File: CS-Examples/11_Formatting/FormatARow.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text
Imports Spire.Xls.Core.Spreadsheet

Namespace FormatARow
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new workbook object
            Dim workbook As New Workbook()

            ' Get the first worksheet in the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Add a new cell style named "newStyle"
            Dim style As CellStyle = workbook.Styles.Add("newStyle")

            ' Set the vertical alignment of the style to center
            style.VerticalAlignment = VerticalAlignType.Center

            ' Set the horizontal alignment of the style to center
            style.HorizontalAlignment = HorizontalAlignType.Center

            ' Set the font color of the style to blue
            style.Font.Color = Color.Blue

            ' Enable the "shrink to fit" option for the style
            style.ShrinkToFit = True

            ' Set the bottom border color of the style to orange-red
            style.Borders(BordersLineType.EdgeBottom).Color = Color.OrangeRed

            ' Set the line style of the bottom border of the style to dotted
            style.Borders(BordersLineType.EdgeBottom).LineStyle = LineStyleType.Dotted

            ' Apply the "newStyle" cell style to all cells in row 1
            sheet.Rows(1).CellStyleName = style.Name

            ' Set the text of all cells in row 1 to "Test"
            sheet.Rows(1).Text = "Test"

            ' Specify the file name for saving the workbook
            Dim result As String = "FormatARow_result.xlsx"

            ' Save the modified workbook to the specified file in Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/11_Formatting/FormatARow.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Xls.Core.Spreadsheet;

namespace FormatARow
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a workbook
            Workbook workbook = new Workbook();
        
            //Get the first sheet
            Worksheet sheet = workbook.Worksheets[0];

            //Create a new style
            CellStyle style = workbook.Styles.Add("newStyle");

            //Set the vertical alignment of the text
            style.VerticalAlignment = VerticalAlignType.Center;

            //Set the horizontal alignment of the text
            style.HorizontalAlignment = HorizontalAlignType.Center;

            //Set the font color of the text
            style.Font.Color = Color.Blue;

            //Shrink the text to fit in the cell
            style.ShrinkToFit = true;

            //Set the bottom border color of the cell to OrangeRed
            style.Borders[BordersLineType.EdgeBottom].Color = Color.OrangeRed;

            //Set the bottom border type of the cell to Dotted
            style.Borders[BordersLineType.EdgeBottom].LineStyle = LineStyleType.Dotted;
           
            //Apply the style to the second row
            sheet.Rows[1].CellStyleName = style.Name;

            sheet.Rows[1].Text = "Test";

            String result = "FormatARow_result.xlsx";

            //Save to file
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //View the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
